wordpress-mobile/WordPress-iOS
# Zendesk Mobile: add Zendesk help options

Fixes #9053 

**NOTES:**
- Right now, this view can only be accessed the 'Help' button on the nav bar. Replacing all instances of the Helpshift `SupportViewController` will come later.
- Right now, the help options only work if you are logged into a WP or gmail account. Support for others is still to come.
- You must have updated your `.wpcom_app_credentials`. If not, new credentials are at https://wp.me/P77Llu-8m1-p2.
- And lastly, the Zendesk views are part of the Zendesk SDK, i.e. I did not create them. As such, if you find any issues feel free to note them but I may not be able to do anything about it.

**To test:**

With Zendesk _enabled_: (the default, assuming the credentials are correct)
- Log in to a WP/social account.
- Press 'Help' on the nav bar (ex: from Site Creation).
- Verify 'WordPress Help Center' displays the Zendesk Guide. (Screenshot below. Yes, those are the only articles right now.)
- Verify 'Contact Us' displays the Zendesk new request. (screenshot below)
- Verify 'My Tickets' displays the Zendesk tickets list. (screenshot below)

You can actually submit tickets (don't worry, they're filtered in Zendesk). Just don't use the email associated with your Zendesk agent account.

With Zendesk _disabled_: (suggestion: return `false` in `ZendeskUtils` > `zendeskEnabled`)
- Log in to a WP/social account.
- Press 'Help' on the nav bar.
- Verify 'WordPress Help Center' displays the WP support page.
- Verify 'WordPress Forums' is shown, and displays the forums page.
- Verify 'Contact Us' is not shown.
- Verify 'My Tickets' is not shown.

**Zendesk Guide:**
![zendesk_guide](https://user-images.githubusercontent.com/1816888/38394222-e1d48a20-38ea-11e8-8fb8-3d6c16c14b4b.png)

**Zendesk New Request:**
![new_request](https://user-images.githubusercontent.com/1816888/38394326-3ef78b1c-38eb-11e8-852b-650b28f8a883.png)

**Zendesk Ticket List:**
![ticket_list](https://user-images.githubusercontent.com/1816888/38394360-64ca4c44-38eb-11e8-9036-81661ae95e28.png)

![tickets](https://user-images.githubusercontent.com/1816888/38394694-c01d3088-38ec-11e8-9a63-3159fb8188d1.png)

@aerych , @etoledom - do either of you have time for a review? Thank you!!
